src/ui/components/AccountSelector.js

```diff
@@ -27,24 +26,46 @@ export default function AccountDropdown({
   placeholder,
   setCurrentIdentity,
   ledger,
+  balance,
 }: DropdownProps) {
   const items = ledger.accounts().filter((a) => a.active);
+  const [selectedAccount, setSelectedAccount] = useState<Account | null>(null);
 
-  const onComboChange = (e, inputObj) => setCurrentIdentity(inputObj);
+  const onComboChange = (e, inputObj) => {
+    setSelectedAccount(inputObj);
+    setCurrentIdentity(inputObj);
+  };
+
+  const renderInput = (params) => {
+    const name = selectedAccount ? selectedAccount.identity.name : "";
+    const balStr = selectedAccount
+      ? ` (${format(selectedAccount.balance, 2)})`
+      : "";
+    return (
+      <TextField
+        fullWidth
+        {...params}
+        inputProps={{...params.inputProps, value: `${name}${balStr}`}}
+        label={placeholder}
+      />
+    );
+  };
 
   return (
     <ThemeProvider theme={theme}>
       <Autocomplete
         onChange={onComboChange}
         fullWidth
         options={items}
-        getOptionLabel={(item) =>
-          `${item.identity.name} (${format(item.balance, 2)})`
-        }
-        style={{margin: "20px"}}
-        renderInput={(params) => (
-          <TextField fullWidth {...params} label={placeholder} />
-        )}
+        getOptionLabel={(item) => {
+          const balStr = format(
+            balance !== fromInteger(0) && balance ? balance : item.balance,
+            2
+          );
+          return `${item.identity.name} (${balStr})`;
```

Comment:
Can we give an explicit type for `item`?

Reply:
yes